### PR TITLE
feat(me): add jsonrpc tests for dbExecute and introspect

### PR DESCRIPTION
**Pull request by jkomyno**
Closes https://github.com/prisma/prisma-engines/issues/3559.

### CODE REVIEW

File: migration-engine/cli/tests/cli_tests.rs [hunk 1]
```diff
@@ -355,6 +356,147 @@ fn introspect_empty_database() {
...
+        .stdin(std::process::Stdio::piped())
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::null())
+        .spawn()
+        .unwrap();
+
+    let stdin = process.stdin.as_mut().unwrap();
+    let mut stdout = BufReader::new(process.stdout.as_mut().unwrap());
+
+    let script = formatdoc! {r#"
```
Comment (pimeys) [inline migration-engine/cli/tests/cli_tests.rs:390]: Could this be something even simpler? Like `SELECT 1`?

File: migration-engine/cli/tests/cli_tests.rs [hunk 2]
```diff
@@ -355,6 +356,147 @@ fn introspect_empty_database() {
     assert!(response.starts_with(r##"{"jsonrpc":"2.0","error":{"code":4466,"message":"An error happened. Check the data field for details.","data":{"is_panic":false,"message":"The introspected database was empty.","meta":null,"error_code":"P4001"}},"id":1}"##));
 }
 
+// TODO: it works fine if run once, it fails with a "Relation already exists" error if run twice in a row.
```
Comment (pimeys) [inline migration-engine/cli/tests/cli_tests.rs:359]: Try with something simpler. Do no modifications to the database, just execute a simple statement. We don't test any schema changes anyhow here, just that we can execute a query.

File: migration-engine/cli/tests/cli_tests.rs [hunk 3]
```diff
@@ -355,6 +356,147 @@ fn introspect_empty_database() {
...
+        }
+    }))
+    .unwrap();
+    stdin.write_all(msg.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+
+    let mut response = String::new();
+    stdout.read_line(&mut response).unwrap();
+
+    assert!(response.starts_with(r##"{"jsonrpc":"2.0","result""##));
```
Comment (pimeys) [inline migration-engine/cli/tests/cli_tests.rs:496]: Could we use `expect_test` snapshot tests here? Would be easier to fix if we change something, and the test output is nicer. 

example: https://github.com/prisma/prisma-engines/blob/main/introspection-engine/introspection-engine-tests/tests/postgres/gin.rs#L38-L49

**Review by pimeys — COMMENTED** [review]
Looks good, maybe think about the few questions I asked here.

**Review by pimeys — APPROVED** [review]
lgtmq

**Comment by tomhoule** [comment]
The tests in migration-engine-cli are really not meant for functional testing, they're sanity checks, we should come up with a different test setup if we want to exercise these commands more end-to-end.

**Pull request merged**